src/table/hooks/use-snapshot.ts

```diff
@@ -0,0 +1,52 @@
+import { onTakeSnapshot } from '@nebula.js/stardust';
+import type { LayoutService, ViewService, SnapshotLayout, HyperCube } from '../../types';
+import findVisibleRows from '../utils/find-visible-rows';
+import { getTotalPosition } from '../../handle-data';
+
+interface UseSnapshotProps {
+  layoutService: LayoutService;
+  viewService: ViewService;
+  model: EngineAPI.IGenericObject | undefined;
+  rootElement: HTMLElement;
+}
+
+const useSnapshot = ({ layoutService, viewService, model, rootElement }: UseSnapshotProps) => {
+  onTakeSnapshot(async (snapshotLayout: SnapshotLayout) => {
+    if (!snapshotLayout.snapshotData) {
+      return snapshotLayout;
+    }
+
+    if (!model) {
+      return snapshotLayout;
+    }
+
+    if (!snapshotLayout.snapshotData.content) {
+      if ((model as EngineAPI.IGenericObject)?.getHyperCubeData) {
+        if (!snapshotLayout.qHyperCube) {
+          snapshotLayout.qHyperCube = {} as HyperCube;
+        }
+        const totalsPosition = getTotalPosition(layoutService.layout);
+        const { visibleRowStartIndex = -1, visibleRowEndIndex = -2 } = findVisibleRows(rootElement, totalsPosition);
+        snapshotLayout.qHyperCube.qDataPages = await (model as EngineAPI.IGenericObject).getHyperCubeData(
```

Comment:
when taking a snapshot using nebula, model is undefined. So it does not work.

Reply:
Therefore, we need to handle qDataPages from layout.qHyperCube like https://github.com/qlik-oss/sn-org-chart/blob/master/src/snapshot.js#L13

Reply:
<img width="628" alt="Screenshot 2023-05-15 at 15 19 26" src="https://github.com/qlik-oss/sn-table/assets/4471489/1ce5eda7-f885-42d7-b7cd-76e74f1ce309">

I had a try

Reply:
I was wrong. Never mind.